src/main/java/com/microsoft/spring/data/gremlin/config/AbstractGremlinConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.config;
+
+import com.microsoft.spring.data.gremlin.common.GremlinFactory;
+import com.microsoft.spring.data.gremlin.conversion.MappingGremlinConverter;
+import com.microsoft.spring.data.gremlin.query.GremlinTemplate;
+import lombok.SneakyThrows;
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.springframework.context.annotation.Bean;
+
+public abstract class AbstractGremlinConfiguration extends GremlinConfigurationSupport {
+
+    public abstract GremlinFactory gremlinFactory();
+
+    @Bean
+    public Client gremlinClient() {
+        return gremlinFactory().getGremlinClient();
+    }
+
+    @Bean
+    @SneakyThrows
+    public MappingGremlinConverter mappingGremlinConverter() {
```

Comment:
Can we do not use the SneakyThrows annotation here? What's the exception will be thrown if there is no this SneakyThrows annotation? 

Per my test:
```Java
    public static void main(String[] args) {
        throwException();
    }

    @SneakyThrows
    public static void throwException() {
        throw new IOException("fake io exception");
    }
```

If the method is annotated with `SneakyThrows` annotation, the caller does not need to handle the thrown `IOException`  at all. 

For the case here, does that mean the user who calls `mappingGremlinConverter` will not know what exception will be thrown at all?

Reply:
yeah, my original prefer to just die here. So export that to user may be a better choice.